### ISSUE TITLE
My test result

### ISSUE THREAD

**Issue by akamob** [EPVelasco/pc2image#1]
Hi, Thank you for the awesome work.

I reproduce your work successfully and the following figure is the result:
![result_1](https://user-images.githubusercontent.com/49118957/147481231-3ef7f3a5-732c-465c-b016-e86146216743.jpg)

The depth image is "unclear" compare to your result (pc2image.GIF).
Did you use sensor fusion? I noticed that your result has a image topic. In other words, the depth image is more clear due to use sensor fusion?

May i have your suggestions to understanding this phenomenon?
Any help is much appreciated:)

**Comment by EPVelasco** [comment]
Hi, thanks for using my code. 
In [pc2 image/ launch/VLP16_image.launch](https://github.com/EPVelasco/pc2image/blob/main/launch/VLP16_image.launch), what value do you have in maxlen?. This value is the maximum range that you want to convert to an image, in the experiment I have put 150 meters 
```
<param name="maxlen"               type="double" value="150" /> 
```
For a room, try to put 20 meters 
```
<param name="maxlen"               type="double" value="20" />  
```

In this code, nearby objects are represented in the image with a value of 65536 and distant objects with a value of 0.

**Comment by akamob** [comment]
Hi, @EPVelasco, thank you very much for your reply.

I’m wondering if it’s possible to extract **specific size** of depth image? such as 512 * 416 or 128 * 128. In other words, I hope these depth images can maintain proper proportions.

I tried some methods:
1. Use opencv or API to resize / crop depth images.
2. Change FOV of VLP-16.
3. Use detector to detect objects I wanted and extract it as depth images.
But these methods cause depth images to blur. Is there some rules or limitions if I want to convert VLP-16’s point cloud into depth image with specific size?

Any suggestions is much appreciated:)

**Comment by EPVelasco** [comment]
Hi, sorry for the late reply.
The VLP-16 sensor only has 16 lines, you can only generate images with a resolution of 16x (360*x_angular_resolution). So if you have x_angular resolution = 0.25, you would have 16x1440 resolution images.

For higher resolution images, one way is to augment the data with interpolation, but I haven't prepared the code for that.

**Comment by akamob** [comment]
Hi, @EPVelasco, thank you very much for your reply:)

Yes, I have recently tried to point cloud interpolation or image inpainting, during the process I found these depth images have 3 channels (I use pillow for checking these depth images). This confuses me, because I think **mono16 (16-bit grayscale image)** only have a one channel. But it also gave me some ideas, I'm wondering if it is possible to treat these depth images as RGB images? namely, use image inpainting on these depth images.

I want to use lidar for activity recognition and it is obvious the point cloud will be very sparse in the vertical direction:
![1](https://user-images.githubusercontent.com/49118957/153750943-3ac28768-7015-48e7-82ad-3b2b2ce1b1bc.png)

If I could use your code to convert point clouds into depth images and then use image inpainting (maybe also use color map to give these images colors), this may make it possible to achieve my goal. I am referencing this repository: https://reurl.cc/nEY4nd, but I'm not sure if my idea is right or not.

May i have your suggestions?
Any help is much appreciated:)
